### DISCUSSION TITLE
有没有可能支持云游戏（PC）

### DISCUSSION THREAD

**Discussion by YipKei** [MaaAssistantArknights/MaaAssistantArknights#1350] [Ideas]
从某个相关事件得到的想法

功能实现的前置需求
---
1. 能够打开一个独立交互的浏览器页面，我看obs可以添加浏览器源，不知道怎么实现的，但理论上存在捕获独立定制浏览器窗口的可能性。
2. 对相应浏览器进行操作，我不清楚可否完全模拟操作，还是必须要前台模拟点击，后者类似于鼠标宏lua脚本的原理，就是实现起来需要一个比较复杂的转换逻辑且不支持后台操作
3. core不影响，准确来说是需要一套特殊的，与adb不一致的交互方式

移动端的云游戏理论上和现在maa的实现方式没啥区别？不需要额外考虑了。

难点：
---
1. 云游戏的图像质量非常差，经常会出现画面撕裂的情况，会对截图识别存在明显影响。
2. 将adb操作指令转译为对浏览器操作的指令

存在这样的需求吗？
---
存在的，很多人的pc性能真的就带不起模拟器，核心原因在于模拟器会提前支取一定的资源，哪怕暂时用不上，这是模拟器的性质决定的，而chrome再怎么内存杀手也比模拟器好多了，MAA core在限制截图频率的前提下，一些基本操作的资源消耗还是比较小的，毕竟绝大多数的消耗都来自于图像识别的过程，降低频率或许会导致部分功能不可用，但可以明显降低消耗。

一些延伸的想法
---
前面提到降低截图频率降低资源开支，那是否可以增加一个节能模式，节能模式下降低截图频率，同时锁死部分需要高频识别的功能，比如基建和自动肉鸽，这样对于低性能PC或者笔记本也可以有一个比较良好的使用体验。

**Comment by aa889788** [comment]
https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/252
